### ISSUE TITLE
Not obvious how to search through documentation

### ISSUE THREAD

**Issue by rotu** [ros-infrastructure/rosindex#232]
There's a lot of great documentation and tutorials on index.ros.org, but it's not as discoverable as it could be. When I'm browsing the docs, the "Search ROS" search box at the top right only searches through packages - it would be nice to have a search box for the documentation and tutorials.

**Comment by clalancette** [comment]
Interestingly, if you type something in that search box and then hit the https://index.ros.org/search URL, you do get the option to search through the documentation and tutorials as well.  So the backend is there, its just presented poorly.  I think this should probably be presented on the front page somehow.

**Comment by hidmic** [comment]
>  When I'm browsing the docs, the "Search ROS" search box at the top right only searches through packages - it would be nice to have a search box for the documentation and tutorials.

Hmm, @rotu can't you select the `documentation` tab after searching? The `lunr.js` based search engine is feature limited, but it's there.

> So the backend is there, its just presented poorly. I think this should probably be presented on the front page somehow.

That's a fair point. Since content is arguably unrelated, at the time I thought a Github search-like UI with categories would work best. I explored putting all content sequentially in a single page and I found it somewhat impractical e.g. either the page layout was too lengthy or each category would show way too few elements to be useful. But this is all subjective.

**Comment by rotu** [comment]
> Hmm, @rotu can't you select the `documentation` tab after searching? The `lunr.js` based search engine is feature limited, but it's there.

Yes! But silly me, I didn't even notice that tab until @clalancette pointed it out!

My mind must have just assumed that was for API documentation pages like http://docs.ros2.org/dashing/api/rclcpp/classrclcpp_1_1Node.html That tab bar was a visual cue that these are "similar" things like the distro selection on a package page.

It might be enough to just have the search results default to the "Documentation" tab when you search from a doc page.

**Comment by tfoote** [comment]
Searching on index.ros.org is now scoped down to just the index. The documentation on docs.ros.org with tutorials has it's own search since we moved content there.
